packages/edit-site/src/components/dataviews/dataviews.js

```diff
@@ -45,7 +45,7 @@ export default function DataViews( {
 			sorting: view.sort
 				? [
 						{
-							id: view.sort.field,
+							id: view.sort.orderby,
```

Comment:
I'm not opinionated about the new name (`orderby` or any alternative), but I think it's best to avoid `field` for clarity.

Reply:
Why is field unclear for you?

Reply:
> Why is field unclear for you?

It suggests the table is sorted by a field (fields API), but it is sorted by the dataset instead.

For example, let's say there was a field called `date` in addition to the `date` property of the dataset. Let's say we have `view.sort.field: date` set. Wouldn't it be confusing, given the view is actually sorted by the _dataset_ `date` and not by the _field_ `date`?

Reply:
More context about accessing the dataset (underlying raw data) vs accessing the fields (cell value) at https://github.com/WordPress/gutenberg/pull/55196#issuecomment-1754970796.

Reply:
I'm not entirely sure what you mean by "dataset". For me it's actually the field. If a dev defines a field that is "full name" that is a computed field from say two properties on the object (first and last name), the sort is the "name" of the field, aka "full name" regardless of how it's really applied under the hood (using two columns to sort for instance)

Reply:
I disagree, for me these are field names. It doesn't matter that right now, we don't define how a field is used for sorting and we just assume that we pass the field name to the REST API, that's a limitation of the current API. long term I expect something like:

```js
const fields = [ {
    name: "fullname",
    data( item ) { return item.firstName + ' ' + item.lastName: }
    sort( query ) {
        query[ 'orderBy' ] = 'firstName,lastName'
    }
} ]
```

and in the view config we'll have something like

```js
sort: { field: 'fullname', direction: 'desc' }
```

Obviously the above is very rough and the "sort" config might be very different.

Reply:
Sorry to jump in... @oandregal Can you explain your thoughts behind calling [`fields` here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/page-pages/index.js#L73) to begin with?

I didn't see anything in https://github.com/WordPress/gutenberg/pull/54966

TanStack uses a `columns` prop and if we want to consider this the JavaScript version of [`WP_List_Table`](https://developer.wordpress.org/reference/classes/wp_list_table/) then I think using `columns` from the start might be clearer?

Reply:
"columns" is specific to table views, and with the same config we want to be able different kind of views starting with table but also rendering things like "grids", "kanban"... So columns doesn't make sense cross "layout". We're basically just defining the "fields" of our data.

Reply:
I agree with @youknowriad here. For us this is a field and is using the `data` internally temporarily.

Reply:
Yeah, I agree `column(s)` is inherently restrictive to a certain view type, but in my opinion `fields(s)` comes with its own connotations, especially in the context of WordPress.

I personally associate it with a single piece of data vs. what has been described as a possible need in the future of combining pieces of data. 

> I'm not opinionated about the new name (orderby or any alternative), **but I think it's best to avoid field for clarity**.

(emphasis mine) I think this also applies to `fields` (plural) as the higher level.

Reply:
I have no other suggestion/aternative, so please don't let this hold progress, but something about `field(s)` seems as restrictive a `column(s)` to me 🤷🏻

Reply:
ok, I see where you are coming from. Let's wrap up this conversation and iterate towards that vision.